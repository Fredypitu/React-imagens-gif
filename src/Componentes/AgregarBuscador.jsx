import { useState } from "react"

export const AgregarBuscador = ({onNewValor})=>{

    const [valorText, setValorText] = useState('');

    const onValorIngresado = ({target:{value}})=>{
        setValorText(value);
    }

    const onSubmit = (event)=>{
        event.preventDefault();
        if (valorText.trim().length<=1) {
            alert('Debe ingresar un valor mayor a 1');
            return;
        };
        // setContArray(cont=>[...cont,valorText]);
        onNewValor(valorText.trim());
        setValorText('');
    }
    return(
        <form onSubmit={(event) => onSubmit(event)}>
            <div>
                <label>
                    Buscar:
                    <input value={valorText} onChange={onValorIngresado} placeholder='Ingrese valor y presione Enter'/>
                </label>
            </div>
    
        </form>

    )
}