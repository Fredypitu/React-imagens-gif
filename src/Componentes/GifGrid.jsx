import { useFetchGifs } from "../Hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGird = ({categoria}) => {
  
 const {imagenes, isLoading} = useFetchGifs(categoria);

 console.log(isLoading);

    return(
        <div>
            <h2>{categoria}</h2>
            {
                    isLoading && (<h2>cargando...</h2>)
            }
     
            {
                imagenes.map((imagen) => (
                    <GifItem key={imagen.id} {...imagen} />
                ))
            }

        </div> 

    )
}