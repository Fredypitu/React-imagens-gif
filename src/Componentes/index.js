
export * from './AgregarBuscador';
export * from './GifGrid';
export * from './GifItem';