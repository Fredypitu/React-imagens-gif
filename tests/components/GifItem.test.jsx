import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/Componentes/GifItem";

describe('Prueba del componente ---> GifItem', () =>{

    const title = 'Prueba';
    const url = 'http://sitema/prueba/imagen.jpga';

    test('Debe recibir los parametros "url" y "title"', () => {
        render(<GifItem title={title} url = {url}/>);
        
        // usando Destructuring
        const {src,alt} = screen.getByRole('img'); // Otra fora ---> screen.getByRole('img').src;  
        expect(src).toBe(url);        
     })

     test('Debe contener el titulo', () => { 
        render(<GifItem title={title} url = {url}/>);
        expect(screen.getByText(title)).toBeTruthy();
      })
})