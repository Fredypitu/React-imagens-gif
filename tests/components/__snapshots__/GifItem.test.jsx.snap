// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente ---> GifItem Debe recibir los parametros "url" y "title" 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Prueba"
      src="http://sitema/prueba/imagen.jpg"
    />
    <p>
      Prueba
    </p>
  </div>
</div>
`;
